sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpInstrumentationPolicy.java

```diff
@@ -249,75 +262,114 @@ public Response<?> process(HttpRequest request, HttpPipelineNextPolicy next) {
                 return null;
             }
 
-            addDetails(request, response, tryCount, span);
-            response = logResponse(logger, response, startNs, requestContentLength, redactedUrl, tryCount,
-                instrumentationContext);
+            addDetails(request, response.getStatusCode(), tryCount, span, metricAttributes);
+            response = logResponse(logger, response, startNs, requestContentLength, redactedUrl, tryCount, context);
             span.end();
             return response;
         } catch (RuntimeException t) {
-            span.end(unwrap(t));
-            // TODO (limolkova) test otel scope still covers this
+            Throwable cause = unwrap(t);
+            if (metricAttributes != null) {
+                metricAttributes.put(ERROR_TYPE_KEY, cause.getClass().getCanonicalName());
+            }
+            span.end(cause);
             throw logException(logger, request, null, t, startNs, null, requestContentLength, redactedUrl, tryCount,
-                instrumentationContext);
+                context);
+        } finally {
+            if (httpRequestDuration.isEnabled()) {
+                httpRequestDuration.record((System.nanoTime() - startNs) / 1_000_000_000.0,
+                    instrumentation.createAttributes(metricAttributes), context);
+            }
         }
     }
 
-    private Span startHttpSpan(HttpRequest request, String sanitizedUrl, InstrumentationContext context) {
-        SpanBuilder spanBuilder = tracer.spanBuilder(request.getHttpMethod().toString(), CLIENT, context)
-            .setAttribute(HTTP_REQUEST_METHOD_KEY, request.getHttpMethod().toString())
-            .setAttribute(URL_FULL_KEY, sanitizedUrl)
-            .setAttribute(SERVER_ADDRESS_KEY, request.getUri().getHost());
-        maybeSetServerPort(spanBuilder, request.getUri());
-        return spanBuilder.startSpan();
+    private void setStartAttributes(HttpRequest request, String sanitizedUrl, SpanBuilder spanBuilder,
```

Comment:
Should we flow through `isTracingEnabled` and `isMetricsEnabled` from the caller? Is it possible for the returns of `tracer.isEnabled()`, `meter.isEnabled()`, and `httpRequestDuration.isEnabled()` to change while instrumentation is running?

If the enabled states are mutable, may I recommend we stick with `disabled` state throughout the call as this could change during the remote execution of the request and we'd be missing data. If that state is enabled at the start of the call we check if it changed to disabled during the remove execution of the request we discard any collected instrumentation.

Reply:
I changed `is*Enabled` to instance vars - we can always make them more dynamic in the future